server/data_common/data_adaptor.py

```diff
@@ -388,3 +388,18 @@ def get_last_mod_time(self):
         except RuntimeError:
             lastmod = None
         return lastmod
+
+
+class AppFeature(object):
+    def __init__(self, path, available=False, method="POST", extra={}):
+        self.path = path
+        self.available = available
+        self.method = method
+        self.extra = extra
+        for k, v in extra.items():
+            setattr(self, k, v)
```

Comment:
```suggestion
        [setattr(self, key, value) for key, value in extra.items()]
```
List comprehension to make it more Pythonic. Also I'd add more descriptive variable names than `key` and `value` once I understand what is actually in `extra`.

Reply:
This looks like some old cruft.  I think everything about "extra" in the AppFeature class can be removed.
I checked and it is not set by the server.  
The AppFeature are used to communicate which endpoints are enabled to the client through the /config:
For example (some output from the config endpoint):
```
    "features": [
      {
        "available": false,
        "method": "POST",
        "path": "/cluster/"
      },
      {
        "available": true,
        "method": "GET",
        "path": "/layout/obs"
      },
```
"extra" was a way to add more key/values to that json blob.  I think diffexp might have set something here at one point.
This kind of pre-dates me.  But in any case we can remove "extra"

Also, a quick check of the client source, and it does not appear that the client even looks at "features".
I think we can safely remove the whole concept.

(As a side note, there are plenty of ways to improve our API)

Reply:
I'm going to remove all of the code related to features in a separate pr just to make the diff clear if anything breaks.